tests/xfstests/partition.pm

```diff
@@ -61,29 +47,115 @@ sub partition_size_num {
+# $filesystem: filesystem type
+# $amount: Amount of partitions to be created for SCRATCH_DEV. Available for btrfs, at most 5.
+# $size: Size of each partition size for TEST_DEV and SCRATCH_DEV. Default: 5120
+# $dev: Optional. Device to be partitioned. Default: same device as root partition
+# $delhome: Delete home partition to get free space for test partition.
+sub do_partition_for_xfstests {
+    my $ref  = shift;
+    my %para = %{$ref};
+    my ($part_table, $part_type, $test_dev);
+    unless ($para{size}) {
+        $para{size} = 5120;
+    }
+    unless ($para{amount}) {
+        $para{amount} = 1;
+    }
+    if ($para{fstype} =~ /btrfs/ && $para{amount} > 5) {
+        $para{amount} = 5;
+    }
+    else {
+        # Mandatory xfs and ext4 has only 1 SCRATCH_DEV
+        $para{amount} = 1;
+    }
+    unless (exists($para{dev})) {
+        my $part = mountpoint_to_partition('/');
+        if ($part =~ /(.*?)(\d+)/) {
+            $para{dev} = $1;
+        }
+    }
+    if (exists($para{delhome}) && $para{delhome} != 0) {
+        my $part = mountpoint_to_partition('/home');
+        remove_partition($part);
+        script_run("sed -i -e '/ \/home /d' /etc/fstab");
+        script_run('mkdir /home/fsgqa; mkdir /home/fsgqa-123456');
+    }
+    parted_print($para{dev});
+    # Prepare suitable partition type, if don't have extended then create one
+    $part_table = partition_table($para{dev});
+    if ($part_table =~ 'msdos') {
+        $part_type = 'logical';
+    }
+    else {
+        $part_type = 'primary';
+    }
+    if ($part_table =~ 'msdos' && partition_num_by_type($para{dev}, 'extended') == -1) {
+        create_partition($para{dev}, 'extended', 'max');
+        parted_print($para{dev});
+    }
+    # Create TEST_DEV
+    $test_dev = create_partition($para{dev}, $part_type, $para{size});
+    parted_print($para{dev});
+    format_partition($test_dev, $para{fstype});
+    # Create SCRATCH_DEV or SCRATCH_DEV_POOL
+    my @scratch_dev;
+    my $num = $para{amount};
+    while ($num != 0) {
+        $num -= 1;
+        my $part = create_partition($para{dev}, $part_type, $para{size});
+        format_partition($part, $para{fstype});
+        push @scratch_dev, $part;
+    }
+    parted_print($para{dev});
+    # Create mount points
+    script_run('mkdir /mnt/test /mnt/scratch');
+    # Generate ~/.xfstests
+    script_run("export TEST_DEV=$test_dev");
+    script_run('export TEST_DIR=/mnt/test');
+    script_run('export SCRATCH_MNT=/mnt/scratch');
+    if ($para{amount} == 1) {
+        script_run("export SCRATCH_DEV=$scratch_dev[0]");
+    }
+    else {
+        script_run("export SCRATCH_DEV_POOL='" . join(' ', @scratch_dev) . "'");
+    }
+    # Sync
+    script_run('sync');
+}
+
 sub run {
     my $self = shift;
     $self->select_serial_terminal;
 
-    # Create partitions
+    # DO NOT set XFSTESTS_DEVICE if you don't know what's this mean
+    # by default we use /home partition spaces for test, and don't need this setting
```

Comment:
Why `/home` ?!
We have to do separate installation with /home only because of xfstests.

Why don't you just simply fix `/usr/share/qa/qa_test_xfstests/partition.py` & `/usr/share/qa/qa_test_xfstests/wrapper.sh` in xfstests ?
Now it became xfstests/openQA abomination which will not "work" without openQA ?
When there is some bug/issue we will first debug if it's openQA issue and then if it's real partition bug ?!

btw it's broken https://openqa.suse.de/tests/4213206#step/run/9

Reply:
> Why `/home` ?!
> We have to do separate installation with /home only because of xfstests.

To use /home partition is to ensure we have enough space, other than root, to do tests.
I didn't change use /home, but from the beginning we use /home, and need a separate installation. I'll try to avoid this in the future, but for now I don't have a good solution.

> 
> Why don't you just simply fix `/usr/share/qa/qa_test_xfstests/partition.py` & `/usr/share/qa/qa_test_xfstests/wrapper.sh` in qa_test_xfstests ?

The scripts you mentioned has no issue to fix, this PR just do what I write in commit. Move two source of code together into one(openQA), because some QA said they confuse by this.

> Now it became xfstests/openQA abomination which will not "work" without openQA ?

QA:Head qa_test_xfstests has no change you can still use it for manually test. This PR only add a ibs way for test, not remove anything. Could you please tell me what specific require to use xfstests? for me actually I think use upstream is better, but it take resources to compiling, so I have to move to ibs. 

> When there is some bug/issue we will first debug if it's openQA issue and then if it's real partition bug ?!

Yes, when we start useing openQA we have to face that. Maybe you didn't met problem before but problem already there, e.g. typing problem in openQA.

> 
> btw it's broken https://openqa.suse.de/tests/4213206#step/run/9

Thank you to point out, let me fix this. This problem seems in installation part.

Reply:
> > Why `/home` ?!
> > We have to do separate installation with /home only because of xfstests.
> 
> To use /home partition is to ensure we have enough space, other than root, to do tests.
> I didn't change use /home, but from the beginning we use /home, and need a separate installation. I'll try to avoid this in the future, but for now I don't have a good solution.

in openQA you can have "any" number of additional disks of any size, why using /home is better solution ? ( # DO NOT set XFSTESTS_DEVICE if you don't know what's this mean) ???

> 
> > Why don't you just simply fix `/usr/share/qa/qa_test_xfstests/partition.py` & `/usr/share/qa/qa_test_xfstests/wrapper.sh` in qa_test_xfstests ?
> 
> The scripts you mentioned has no issue to fix, this PR just do what I write in commit. Move two source of code together into one(openQA), because some QA said they confuse by this.
> 
> > Now it became xfstests/openQA abomination which will not "work" without openQA ?
> 
> QA:Head qa_test_xfstests has no change you can still use it for manually test. This PR only add a ibs way for test, not remove anything. Could you please tell me what specific require to use xfstests? for me actually I think use upstream is better, but it take resources to compiling, so I have to move to ibs.

Thant's the point, why don't keep package which will work on any architecture, openQA or directly in ibs ? Now we have two projects, I guess somebody has to maintain xfstests too to be up to date ?

> 
> > When there is some bug/issue we will first debug if it's openQA issue and then if it's real partition bug ?!
> 
> Yes, when we start useing openQA we have to face that. Maybe you didn't met problem before but problem already there, e.g. typing problem in openQA.

No, we don't have to create unportable something. openQA can run anything, without rewriting part of it into openQA.

> 
> > btw it's broken https://openqa.suse.de/tests/4213206#step/run/9
> 
> Thank you to point out, let me fix this. This problem seems in installation part.

Reply:
I fixed installation in #10222. In this PR, I forgot to check install dir when change to use Petr's repo in QA:Head, so cause your problem. Now it should works.
I don't think more discuss help, you could use many ways to do the same thing. This change simply benefit openQA resources.